bin/runtime/src/lib.rs

```diff
@@ -272,11 +272,21 @@ parameter_types! {
     pub MaximumMultiplier: Multiplier = Bounded::max_value();
 }
 
+pub struct DivideFeeBy<const N: Balance>;
+
+impl<const N: Balance> WeightToFee for DivideFeeBy<N> {
+    type Balance = Balance;
+
+    fn weight_to_fee(weight: &Weight) -> Self::Balance {
+        Balance::saturated_from(weight.ref_time()).saturating_div(N)
+    }
+}
+
 impl pallet_transaction_payment::Config for Runtime {
     type RuntimeEvent = RuntimeEvent;
     type OnChargeTransaction = CurrencyAdapter<Balances, EverythingToTheTreasury>;
-    type LengthToFee = IdentityFee<Balance>;
-    type WeightToFee = IdentityFee<Balance>;
+    type LengthToFee = DivideFeeBy<10>;
```

Comment:
Yes indeed, which maybe is a bit surprising, but since fee is computed as follows

```
fee = weight_fee * next_fee_multiplier + len_fee + base_fee + tip
```

so this `len_fee` can stay as it is as `IdentityFee<Balance>`, does not matter much for decreasing fee overall, as majority of tx fee comes from weight and adaptive fee mechanism

Btw `IdentityFee` code implementation returns `Balance::saturated_from(weight.ref_time())`, so essentially it converts `u64` to `u128` using `saturated_from` method. I could not find it's implementation in source code (it's macro-generated), yet I checked running some unit test that it is 1:1 in most cases e.g.:
```
        let balance: u128 = u128::saturated_from(1_u64);
        assert_eq!(balance, 1_u128);
```

Reply:
Can we check `LengthToFee` in other production substrate runtimes? Because from what you are saying it doesn't make sense in the current form.

Reply:
I mean, we could just as well output `0` here.

Reply:
You're right, on Polkadot Mainnet, encoded ext length is multiplied by a [constant](https://github.com/polkadot-fellows/runtimes/blob/v1.1.3/relay/polkadot/src/lib.rs#L413):
```
	pub const UNITS: Balance = 10_000_000_000;
	pub const DOLLARS: Balance = UNITS; // 10_000_000_000
	pub const GRAND: Balance = DOLLARS * 1_000; // 10_000_000_000_000
	pub const CENTS: Balance = DOLLARS / 100; // 100_000_000
	pub const MILLICENTS: Balance = CENTS / 1_000; // 100_000
[...]
	pub const TransactionByteFee: Balance = 10 * MILLICENTS;
[...]
	type LengthToFee = ConstantMultiplier<Balance, TransactionByteFee>;
```
which means legth fee is `length(ext.encoded()) * 0.0001 DOT` on Polkadot

Reply:
Yeah, question is now if we intended it like this (I doubt it), or was it a mistake.

Reply:
It was last changed:
* [here](https://github.com/Cardinal-Cryptography/aleph-node/commit/aff164faad6628292341f284ecd16367dc497169#diff-3c96b927be822588b1e81d59b5b6a036db0aed737547abbad61bb1a90f94ff08L283), but this was a no-op - this length fee was computed in substrate 0.9.13 as `let length_fee = TransactionByteFee::get() * (xt.clone().encode().len() as Balance);`
* then `TransactionByteFee` = `1` was introduced in the [first version of aleph-runtime](https://github.com/Cardinal-Cryptography/aleph-node/commit/89b1cf167#diff-3c96b927be822588b1e81d59b5b6a036db0aed737547abbad61bb1a90f94ff08), so it was there from the very beginning

Reply:
Created task for it